platforms/r3-corda-ent/charts/cenm-gateway/templates/statefulset.yaml

```diff
@@ -0,0 +1,130 @@
+#
+#  SPDX-License-Identifier: Apache-2.0
+##############################################################################################
+
+---
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "gateway.fullname" . }}
+  namespace: {{ .Release.Namespace }}  
+  labels:
+    app: {{ include "gateway.fullname" . }}
+    app.kubernetes.io/name: gateway-statefulset
+    app.kubernetes.io/component: gateway
+    app.kubernetes.io/part-of: {{ include "gateway.fullname" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/namespace: {{ .Release.Namespace }}
+    app.kubernetes.io/release: {{ .Release.Name }}
+spec:
+  replicas: 1
+  podManagementPolicy: OrderedReady
+  updateStrategy:
+    type: RollingUpdate
+  selector:
+    matchLabels:
+      app: {{ include "gateway.fullname" . }}
+      app.kubernetes.io/name: gateway-statefulset
+      app.kubernetes.io/component: gateway
+      app.kubernetes.io/part-of: {{ include "gateway.fullname" . }}
+      app.kubernetes.io/managed-by: {{ .Release.Service }}
+      app.kubernetes.io/namespace: {{ .Release.Namespace }}
+      app.kubernetes.io/release: {{ .Release.Name }}
+  serviceName: {{ include "gateway.fullname" . }}
+  volumeClaimTemplates:
+    - metadata:
+        name: gateway-logs
+      spec:
+        storageClassName: storage-{{ .Release.Name }}
+        accessModes: [ "ReadWriteOnce" ]
+        resources:
+          requests:  
+            storage: {{ .Values.storage.size }}
+  template:
+    metadata:
+      labels:
+        app: {{ include "gateway.fullname" . }}
+        app.kubernetes.io/name: gateway-statefulset
+        app.kubernetes.io/component: gateway
+        app.kubernetes.io/part-of: {{ include "gateway.fullname" . }}
+        app.kubernetes.io/managed-by: {{ .Release.Service }}
+        app.kubernetes.io/namespace: {{ .Release.Namespace }}
+        app.kubernetes.io/release: {{ .Release.Name }}
+    spec:
+      serviceAccountName: {{ .Values.global.serviceAccountName }}
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 1000
+        fsGroup: 1000
+      hostname: {{ .Release.Name }}
+      imagePullSecrets:
+      {{- if .Values.image.pullSecret }}
+        - name: {{ .Values.image.pullSecret }}
+      {{- end }}
+      initContainers:
+      - name: init-check-auth
+        env:
+          - name: ACCEPT_LICENSE
+            value: "YES"
+        image: {{ .Values.image.gateway.repository }}:{{ .Values.image.gateway.tag }}
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        command: ["/bin/bash", "-c"]
+        args:
+        - |-
+          echo Probing auth.{{ .Release.Namespace }} port {{ .Values.global.cenm.auth.port }}
+          until nc -w 5 -vz auth.{{ .Release.Namespace }} {{ .Values.global.cenm.auth.port}} > /dev/null 2>&1
+          do
+            echo "Connect to auth.{{ .Release.Namespace }} port {{ .Values.global.cenm.auth.port }} (tcp) failed: Connection refused"
```

Comment:
is auth service always going to be called "auth"?

Reply:
Yeah, there were tradeoffs of using release name and fixed name, for the CENM services, For now the auth and the other CENM services having fixed name made things simple.